server/src/main/java/org/elasticsearch/common/time/DateUtils.java

```diff
@@ -293,6 +293,17 @@ public static long toMilliSeconds(long nanoSecondsSinceEpoch) {
         return nanoSecondsSinceEpoch / 1_000_000;
     }
 
+    public static int compareNanosToMillis(long nanos, long millis) {
```

Comment:
Is this comparing sub-second durations, or epoch times? I think it's the former, but please add javadocs to make clear the expectations for input and output.

Reply:
Javadoc added.  The intent is in fact to compare epoch dates in the two different formats.  The intent here is to support an ES|QL use case comparing fields of different types.  This seemed like a good place to put it, since we have a bunch of other functions for dealing with ES date_nanos here, but I can move it if this class isn't appropriate.